src/framework/preferences/js/Store.js

```diff
@@ -131,7 +131,11 @@ var fluid_2_0 = fluid_2_0 || {};
     });
 
     fluid.tempStore.set = function (settings, applier) {
-        applier.requestChange("", settings);
+        if ($.isEmptyObject(settings)) {
+            applier.fireChangeRequest({path: "", type: "DELETE"});
+        } else {
+            applier.change("", settings);
```

Comment:
As per the previous discussion, we should do the DELETE first _every_ time, not just if the model is empty - it might have shrunk for some other reason other than being entirely empty. As a general "code smells" note, any call to $.isEmptyObject needs to be flagged up as suspicious - I've almost never seen a call to it that was really necessary. Although KASPARNET did manage to convince me of one in the GPII once.